cli/inputhandler/wrappers/transaction_input_wrapper.h

```diff
@@ -46,54 +43,49 @@ namespace wrapper
 class TransactionInputWrapper : public IWrapper
 {
 public:
-    explicit TransactionInputWrapper(std::map<uint32_t, std::string> const &inputData) :
+    explicit TransactionInputWrapper(cxxopts::ParseResult const  &inputData) :
             IWrapper(inputData)
-    {
-        m_containsData = (getInputData().find(ARGS_TX_IDX_DATA) != getInputData().end());
-        m_containsReceiverName = (getInputData().find(ARGS_TX_IDX_RECEIVER_NAME) != getInputData().end());
-        m_containsSenderName = (getInputData().find(ARGS_TX_IDX_SENDER_NAME) != getInputData().end());
-        m_containsOptions = (getInputData().find(ARGS_TX_IDX_OPTIONS) != getInputData().end());
-        m_containsChainID = (getInputData().find(ARGS_TX_IDX_CHAIN_ID) != getInputData().end());
-    }
+    {}
 
     uint64_t getNonce() const
     {
-        return std::stoul(getInputData().at(ARGS_TX_IDX_NONCE));
+        return getInputData()["nonce"].as<uint64_t>();
     }
 
     std::string getValue() const
     {
-        return getInputData().at(ARGS_TX_IDX_VALUE);
+        return getInputData()["value"].as<std::string>();
     }
 
     Address getReceiver() const
     {
-        return Address(getInputData().at(ARGS_TX_IDX_RECEIVER));
+        std::string const address = getInputData()["receiver"].as<std::string>();
+        return Address(address);
     }
 
     std::shared_ptr<bytes> getReceiverName() const
     {
-        return internal::getUserInputIfExists<bytes>(getInputData(), ARGS_TX_IDX_RECEIVER_NAME, m_containsReceiverName);
+        return internal::getUserInputIfExists<bytes>(getInputData(), "receiver-name");
```

Comment:
Not sure, is this correct? Above you had `sender-username` & `receiver-username`. I think we can define the keys as constants, but it might be an overkill, so it can stay as it is :)

Reply:
Good finding. It was wrong, I had it mixed it up, so I fixed it.

> I think we can define the keys as constants, but it might be an overkill, so it can stay as it is :)

This was also my first though: make them constants. But then I realized that we will only have few CLI functionalities, so might be indeed an overkill.